src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -50,15 +50,14 @@
  * <p>This class is (outside of tests) a singleton instance, living in `BazelRepositoryModule`.
  */
 public class HttpDownloader implements Downloader {
-  public static final int DEFAULT_MAX_PARALLEL_DOWNLOADS = 8;
   private static final Clock CLOCK = new JavaClock();
   private static final Sleeper SLEEPER = new JavaSleeper();
   private static final Locale LOCALE = Locale.getDefault();
 
   private final Semaphore semaphore;
   private float timeoutScaling = 1.0f;
   private int maxAttempts = 0;
-  private int maxParallelDownloads = DEFAULT_MAX_PARALLEL_DOWNLOADS;
+  private int maxParallelDownloads = 8;
```

Comment:
One more idea that I haven't fully thought through:
1. Make the `Semaphore` field non-final.
2. Move its initialization into `setMaxParallelDownloads`.
3. Get rid of this field entirely.

This is only possible if we expect `setMaxParallelDownloads` to be called in all cases before a download, but I hope that's how things are.